sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -1876,3 +1876,38 @@ case class Decade(child: Expression) extends UnaryExpression with ImplicitCastIn
     defineCodeGen(ctx, ev, c => s"$dtu.getDecade($c)")
   }
 }
+
+case class DatePart(field: Expression, source: Expression, child: Expression)
```

Comment:
As you see the UT failure, I enforced the function description by UT, @MaxGekk . You shall not pass. :grin:
Please add a proper description annotation.

Reply:
I added the description. Just in case, can I run the check locally and get the error earlier?

Reply:
Of course, you can run SQL module test always when you are writing SQL PR.

Reply:
I ran new test `date_part.sql` via:
```shell
build/sbt "sql/test-only *SQLQueryTestSuite -- -z date_part.sql"
```
but didn't get the error.

Reply:
@MaxGekk . 
- First of all, where did you get that idea about that CLI command? The failed test case in Jekins is [SQLQuerySuite.SPARK-14415](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108938/testReport/org.apache.spark.sql/SQLQuerySuite/SPARK_14415__All_functions_should_have_own_descriptions/), not `SQLQueryTestSuite`.
- Second, what I recommended was the SQL **module** test is `build/sbt "project sql" test`. 

Since you are one of the active SQL-area contributors in Apache Spark community, let me recommend you once more to run **the module test** at least before making a PR.
- If you touch `catalyst` module, you should run `build/sbt "project catalyst" test` at least. 
- if you touch `sql` module, you should run `build/sbt "project sql" test` at least.

Reply:
Thank you for the recommendations. I will do that next time.